ipfs/http_api.go

```diff
@@ -0,0 +1,58 @@
+package ipfs_filestore
+
+import (
+	"fmt"
+
+	ipfs_config "gx/ipfs/QmPEpj17FDRpc7K1aArKZp3RsHtzRMKykeK9GVgn4WQGPR/go-ipfs-config"
+	ipfs_commands "gx/ipfs/QmUJYo4etAQqFfSS2rarFAE97eNGB8ej64YkRT2SmsYD4r/go-ipfs/commands"
+	ipfs_core "gx/ipfs/QmUJYo4etAQqFfSS2rarFAE97eNGB8ej64YkRT2SmsYD4r/go-ipfs/core"
+	ipfs_corehttp "gx/ipfs/QmUJYo4etAQqFfSS2rarFAE97eNGB8ej64YkRT2SmsYD4r/go-ipfs/core/corehttp"
+)
+
+// serveAPI makes an IPFS node available over an HTTP api
+func (fs *Filestore) serveAPI() error {
+	if fs.node == nil {
+		return fmt.Errorf("node is required to serve IPFS HTTP API")
+	}
+
+	cfg := fs.cfg
+	addr := ""
+	if cfg.Repo != nil {
+		if ipfscfg, err := cfg.Repo.Config(); err == nil {
+			// TODO (b5): apparantly ipfs config supports multiple API multiaddrs?
+			// I dunno, for now just go with the most likely case of only assigning
+			// an address if one string is supplied
+			if len(ipfscfg.Addresses.API) == 1 {
```

Comment:
Trying to follow this down the rabbit hole, but not totally sure. What happens if we get to:

`ipfs_corehttp.ListenAndServe(fs.node, addr, opts...)`

and `addr == ""` ?

Can't tell if that is an error, because there are a few routes to getting there - if the repo == nil and there is more then one api multiaddress. But I don't know/understand if that is okay or what the behavior is or what should be done about it if anything.